Documentation/ceph-csi-drivers.md

```diff
@@ -0,0 +1,281 @@
+---
+title: Ceph CSI
+weight: 28
+indent: true
+---
+
+# Running ceph CSI drivers with Rook
+
+Here is a guide on how to implement ceph-csi drivers on a Kubernetes cluster with Rook. The way to implement it should change relatively soon as there is an effort to implement a csi plugin within Rook directly ([#1385](https://github.com/rook/rook/issues/1385), [#2059](https://github.com/rook/rook/pull/2059)).
+
+- [Enable CSI drivers](#enable-csi-drivers)
+- [Test the CSI driver](#test-the-csi-driver)
+- [Snapshots](#snapshots)
+- [Cleanup](#cleanup)
+
+## Prerequisites
+
+1. A Kubernetes v1.12+ cluster with at least one node
+2. `--allow-privileged` flag set to true in [kubelet](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/) and your [API server](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/)
+3. An up and running Rook instance (see [Rook - ceph quickstart guide](https://github.com/rook/rook/blob/master/Documentation/ceph-quickstart.md))
+4. Make sure the required feature gates as stated in the [kubernetes-csi setup guide](https://kubernetes-csi.github.io/docs/Setup.html) are enabled in kubelet. Assuming you're running Kubernetes v1.12.x, you should enable these feature gates: `CSINodeInfo`, `CSIDriverRegistry`, `VolumeSnapshotDataSource`. For other versions, refer to the [feature gates documentation](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/#overview)
```

Comment:
@rootfs If we have to enable any feature gates, how can we rely on the csi driver? Since CSI itself doesn't need a feature gate, is there a mode for CSI where we don't need these enabled?

Reply:
The above feature gates are [CSI doc requirements](https://kubernetes-csi.github.io/docs/Setup.html), they are not enabled by default (at least 1.12 and 1.13). 

`VolumeSnapshotDataSource` is for snapshot use case. rbd csi driver supports snapshot.

For ceph-csi, I haven't hit issues with missing `CSINodeInfo` or `CSIDriverRegistry` features. @mickymiek do you have any special cases that need these features?